internal/databases/mysql/stream_fetch_handler.go

```diff
@@ -130,8 +141,9 @@ func fetchBinlogs(folder storage.Folder, sentinel StreamSentinelDto, binlogsAreD
 	binlogsAreDone <- nil
 }
 
-func BinlogShouldBeFetched(sentinel StreamSentinelDto, binlogName string, endTS *time.Time, object storage.Object) bool {
-	return sentinel.BinLogStart <= binlogName && (endTS == nil || (*endTS).After(object.GetLastModified()))
+func BinlogShouldBeFetched(backupStartUploadTime time.Time, endTS *time.Time, object storage.Object) bool {
+	return (backupStartUploadTime.Before(object.GetLastModified()) || backupStartUploadTime.Equal(object.GetLastModified())) &&
+		(endTS == nil || (*endTS).After(object.GetLastModified()))
```

Comment:
Suppose we have binlog `my-bin.0042` started at 12:00, finished at 14:00 and uploaded to S3 at 14:05
User ask for PITR at 13:00 - so we need to fetch this binlog, even if it's upload date after `endTS`

Reply:
It will be fixed in another PR.